src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+            allowMeasureColumns: true,
+            allowTimeSeriesColumns: false,
+          },
+          {
+            key: "legend",
+            label: "Slice with color",
+            allowAttributeColumns: true,
+            allowMeasureColumns: false,
+            allowTimeSeriesColumns: false,
+          },
+        ],
+      },
+    ],
+    visualPropEditorDefinition: {
+      elements: [
+        {
+          key: "datalabels",
+          type: "toggle",
+          defaultValue: false,
+          label: "Data Labels",
+        },
+        {
+          key: "hideLegend",
+          type: "toggle",
+          defaultValue: true,
+          label: "Legends",
+        },
+      ],
+    },
+    validateConfig: (updatedConfig, chartModel): ValidationResponse => {
+      if (updatedConfig.length <= 0) {
+        return {
+          isValid: false,
+          validationErrorMessage: ["invalid config. no config found"],
+        };
+      }
+      // assuming 0 is x dimension
+      const xAxisDimensions = updatedConfig[0];
+      const yAxisDimensions = updatedConfig
+        .map((config) => ({ type: config.key, dimensions: config.dimensions }))
+        .map((item) => {
+          const filteredColumns = item.dimensions.filter(
+            (column) => column.key === "y"
+          );
+          return { axis: filteredColumns[0], type: item.type };
+        });
+
+      const legendValidation = () => {
+        const res = {
+          isValid: true,
+          errorMessage: undefined,
+        };
+
+        for (const chart of updatedConfig) {
+          const legend = chart.dimensions.find((dim) => dim.key === "legend");
+
+          if (legend && legend.columns.length > 0) {
+            // Check if "y" axis column is not empty
+            const yDimension = chart.dimensions.find((dim) => dim.key === "y");
+
+            if (yDimension) {
+              if(yDimension.columns.length === 0) {
+                res.isValid = false; 
+                res.errorMessage = `Invalid config. Y axis column should not be empty for ${chart.key} chart while slicing with an attribute`;
+                return res;
+              }
+              if(yDimension.columns.length > 1) {
+                res.isValid = false; 
+                res.errorMessage = `Invalid config. Y axis column should not be more than 1 for ${chart.key} chart while slicing with an attribute`;
+                return res;
+              }
+            }
+          }
+        }
+        return res;
+      };
+
+      const xAxisValidation = () => {
+        if (xAxisDimensions.dimensions[0].columns.length === 0) {
+          return {
+            isValid: false,
+            errorMessage: "Invalid config. X axis columns cannot be empty",
+          };
+        } else {
+          return {
+            isValid: true,
+          };
+        }
+      };
+
+      const yAxisValidation = () => {
+        if (find(yAxisDimensions, (axis) => axis.axis.columns.length !== 0)) {
```

Comment:
this will pass even if one config is valid... have we checked this?

Reply:
It won't. I have verified